scripts/exasol_xmlrpc.py

```diff
@@ -1,65 +1,94 @@
-    max_iterations = WAIT_MAX_ITERATIONS
 
-    while max_iterations > 0:
-        print("Waiting iteration count: %s" % max_iterations)
+def wait(server: ServerProxy) -> None:
+    start = datetime.datetime.now()
+
+    def current_delta():
+        return datetime.datetime.now() - start
+
+    def remaining():
+        return WAIT_DURATION - current_delta()
+
+    while remaining().total_seconds() > 0:
+        logger.info(
+            f"Waiting for {WAIT_DURATION} (remaining: {remaining()}) with sleep time {WAIT_SLEEP_TIME}s"
+        )
 
-        started = has_started(server)
-        if started:
-            break
+        if has_started(server):
+            logger.info(f"Server started after {current_delta()}")
+            return
 
-        max_iterations -= 1
-        time.sleep(sleep_time)
+        time.sleep(WAIT_SLEEP_TIME.seconds)
 
-    if max_iterations == 0:
-        raise Exception("Exasol management node could not be started after long time!")
+    if remaining().total_seconds() < 0:
+        raise TimeoutError(
+            f"Exasol management node could not be started after {current_delta()}!"
+        )
 
-def create_server(address, username, password):
-    url = CLUSTER_URL.format(username, password, address)
-    server = xmlrpc(url, context=ssl._create_unverified_context())
-    return server
 
-def check_db_started(server):
-    print("Checking if 'exadb' database is running")
-    if not server.db_exadb.runningDatabase():
-        raise Exception("Exasol database is not started!")
+def create_server(address, username, password) -> ServerProxy:
+    url = f"https://{username}:{password}@{address}/cluster1"
+    logger.info(f"Connecting to {username} @ {address}")
+    return xmlrpc(url, context=ssl._create_unverified_context())
+
+
+def check_db_started(server: ServerProxy):
+    logger.info("Checking if 'exadb' database is running")
+    running = server.db_exadb.runningDatabase()
+    if running:
+        logger.info(f"Exasol database is running: {running}")
+    else:
+        raise RuntimeError("Exasol database is not started!")
+
 
 def run():
-    parser = argparse.ArgumentParser(description='Exasol XMLRPC Interactions')
-    parser.add_argument('--license-server-address', type=str, required=True)
-    parser.add_argument('--username', type=str, required=True)
-    parser.add_argument('--password', type=str, required=True)
+    parser = argparse.ArgumentParser(description="Exasol XMLRPC Interactions")
+    parser.add_argument("--license-server-address", type=str, required=True)
+    parser.add_argument("--username", type=str, required=True)
+    parser.add_argument("--password", type=str, required=True)
 
     args = parser.parse_args()
-    print("The following arguments are provided: '%s'" % args)
+    logger.info("The following arguments are provided: '%s'" % args)
 
     try:
-        server = create_server(args.license_server_address, args.username, args.password)
+        server = create_server(
```

Comment:
Even though the class name of he returned object  is `ServerProxt`, semantically for me it looks more like a client, so
```suggestion
        client = create_client(
```